src/commonMain/kotlin/exchange.dydx.abacus/output/account/SubaccountFundingPayment.kt

```diff
@@ -17,66 +15,21 @@ data class SubaccountFundingPayment(
-            parser: ParserProtocol,
-            data: Map<*, *>?,
-        ): SubaccountFundingPayment? {
-            Logger.d { "creating Account Funding Payment\n" }
-
-            data?.let {
-                val marketId = parser.asString(data["marketId"])
-                val payment = parser.asDouble(data["payment"])
-                val rate = parser.asDouble(data["rate"])
-                val positionSize = parser.asDouble(data["positionSize"])
-                val price = parser.asDouble(data["price"])
-                val effectiveAtMilliSeconds =
-                    parser.asDatetime(data["effectiveAt"])?.toEpochMilliseconds()?.toDouble()
-                if (marketId != null && payment != null && rate != null && positionSize != null && effectiveAtMilliSeconds != null) {
-                    return if (existing?.marketId != marketId ||
-                        existing.payment != payment ||
-                        existing.rate != rate ||
-                        existing.positionSize != positionSize ||
-                        existing.price != price ||
-                        existing.effectiveAtMilliSeconds != effectiveAtMilliSeconds
-                    ) {
-                        SubaccountFundingPayment(
-                            marketId,
-                            payment,
-                            rate,
-                            positionSize,
-                            price,
-                            effectiveAtMilliSeconds,
-                        )
-                    } else {
-                        existing
-                    }
-                }
-            }
-            Logger.d { "Account Funding Payment not valid" }
-            return null
-        }
-
-        fun create(
+        internal fun merge(
             existing: IList<SubaccountFundingPayment>?,
-            parser: ParserProtocol,
-            data: List<Map<String, Any>>?,
-        ): IList<SubaccountFundingPayment>? {
-            return ParsingHelper.merge(parser, existing, data, { obj, itemData ->
-                val time1 = (obj as SubaccountFundingPayment).effectiveAtMilliSeconds
-                val time2 =
-                    parser.asDatetime(itemData["effectiveAt"])?.toEpochMilliseconds()
-                        ?.toDouble()
-                ParsingHelper.compare(time1, time2 ?: 0.0, true)
-            }, { _, obj, itemData ->
-                obj ?: SubaccountFundingPayment.create(
-                    null,
-                    parser,
-                    parser.asMap(itemData),
-                )
-            })?.toIList()
+            new: IList<SubaccountFundingPayment>?,
+        ): IList<SubaccountFundingPayment> {
+            return ParsingHelper.merge(
+                existing = existing,
+                new = new,
+                comparison = { obj, newItem ->
+                    ParsingHelper.compare(obj.effectiveAtMilliSeconds, newItem.effectiveAtMilliSeconds, true)
```

Comment:
I don't think `effectiveAtMilliSeconds` exist in the Indexer response. Are you creating this key later in the pipeline?